### PR TITLE
Apply caching to filter autocomplete dropdown

### PR DESCRIPTION
**Pull request by vnnw**
Implement: Cache endpoint by datasource and column.

### CODE REVIEW

File: superset/views/core.py [hunk 1]
```diff
@@ -1275,13 +1276,44 @@ def filter(self, datasource_type, datasource_id, column):
...
-
-        payload = json.dumps(
-            datasource.values_for_column(
-                column,
-                config.get('FILTER_SELECT_ROW_LIMIT', 10000),
-            ),
-            default=utils.json_int_dttm_ser)
+        # Implement: Cache endpoint by datasource and column
+        cache_key = hashlib.md5((datasource_id + column).encode('utf-8')).hexdigest()
+        if cache_key and cache:
```
Comment (xrmx) [inline superset/views/core.py]: cache_key will always evaluate True

File: superset/views/core.py [hunk 2]
```diff
@@ -1275,13 +1276,44 @@ def filter(self, datasource_type, datasource_id, column):
...
+            if (hasattr(datasource, 'database') and datasource.database.cache_timeout):
```
Comment (xrmx) [inline superset/views/core.py]:
```
cache_timeout = config.get('CACHE_DEFAULT_TIMEOUT')
try:
   if datasource.database.cache_timeout:
       cache_timeout = ...
except AttributeError:
   if datasource.cache_timeout:
       cache_timeout = ...
```

File: superset/views/core.py [hunk 3]
```diff
@@ -1275,13 +1276,44 @@ def filter(self, datasource_type, datasource_id, column):
...
+                    config.get('FILTER_SELECT_ROW_LIMIT', 10000),
+                )
+                try:
+                    payload = json.dumps(cache_value, default=utils.json_int_dttm_ser)
+                    stats_logger.incr('set_cache_key')
+                    cache.set(cache_key, cache_value, cache_timeout)
+                except Exception as e:
+                    logging.warning('Error writing cache {}:'.format(cache_key))
+                    logging.exception(e)
+                    cache.delete(cache.key)
```
Comment (xrmx) [inline superset/views/core.py]: cache_key

File: superset/views/core.py [hunk 4]
```diff
@@ -1275,13 +1276,44 @@ def filter(self, datasource_type, datasource_id, column):
...
+        cache_key = hashlib.md5((datasource_id + column).encode('utf-8')).hexdigest()
+        if cache_key and cache:
+            if (hasattr(datasource, 'database') and datasource.database.cache_timeout):
+                cache_timeout = datasource.database.cache_timeout
+            elif (hasattr(datasource, 'database') and datasource.cache_timeout):
+                cache_timeout = datasource.database.cache_timeout
+            else:
+                cache_timeout = config.get('CACHE_DEFAULT_TIMEOUT')
+            cache_value = cache.get(cache_key)
+            if cache_value:
```
Comment (xrmx) [inline superset/views/core.py]: Looks like of duplicated code here

File: superset/views/core.py [hunk 5]
```diff
@@ -1275,13 +1276,44 @@ def filter(self, datasource_type, datasource_id, column):
...
+            else:
+                cache_value = datasource.values_for_column(
+                    column,
+                    config.get('FILTER_SELECT_ROW_LIMIT', 10000),
+                )
+                try:
+                    payload = json.dumps(cache_value, default=utils.json_int_dttm_ser)
+                    stats_logger.incr('set_cache_key')
+                    cache.set(cache_key, cache_value, cache_timeout)
+                except Exception as e:
```
Comment (xrmx) [inline superset/views/core.py]: You don't know if it was writing the cache that failed

File: superset/views/core.py (hunk 2 above)
Comment (vnnw) [reply superset/views/core.py]: I don't think using try-except is proper here. There are three possible places to get cache_timeout value. I wanted to make it cascading fallback.

Comment (xrmx) [reply superset/views/core.py]: Updated the code above

File: superset/views/core.py (hunk 4 above)
Comment (mistercrunch) [reply superset/views/core.py]: Agreed, I like the idea of caching the filters autocomplete, but we need to refactor some of the caching logic in a single place.

**Comment by mistercrunch** [comment]
Title of the PR is not reflective of content, re-titling it.

**Pull request closed without merging**